### PR TITLE
add network_ids, network_count, available_networks

### PR DESCRIPTION
**Pull request by cuteolaf**

### CODE REVIEW

**Comment by Szegoo** [comment]
@cuteolaf Storing the network ids in a vector is storage expensive. I have made changes so that we again have `network_id_count`. Let me know if this is good for you.

File: contracts/identity/lib.rs [hunk 1]
```diff
@@ -372,6 +373,19 @@ mod identity {
 			Ok(())
 		}
 
+		#[ink(message)]
+		pub fn network_count(&self) -> u32 {
+			self.network_id_count
+		}
```
Comment (Szegoo) [inline contracts/identity/lib.rs]: We probably don't need this
```suggestion
```

File: contracts/identity/tests.rs [hunk 2]
```diff
@@ -352,6 +354,9 @@ fn remove_network_works() {
 
 	assert!(identity.network_name_of.get(0).is_none());
 
+	// assert_eq!(identity.network_count(), 0);
```
Comment (Szegoo) [inline contracts/identity/tests.rs]:
```suggestion
```

**Comment by cuteolaf** [comment]
```rust
(0..self.network_id_count)
		.map(|id| (id, self.network_name_of(id)))
		.filter_map(|(id, maybe_network)| maybe_network.map(|name| (id, name)))
		.collect()
```

I'm not happy with this implementation.

**Review by Szegoo — APPROVED** [review]

**Pull request merged**